lib/chartmogul/api_resource.rb

```diff
@@ -31,14 +32,8 @@ def self.set_resource_root_key(root_key)
     end
 
     def self.connection
-      @connection ||= Faraday.new(url: ChartMogul::API_BASE) do |faraday|
-        faraday.use Faraday::Request::BasicAuthentication, ChartMogul.account_token, ChartMogul.secret_key
-        faraday.use Faraday::Response::RaiseError
-        faraday.request :retry, max: ChartMogul.max_retries, retry_statuses: RETRY_STATUSES,
-                                max_interval: MAX_INTERVAL, backoff_factor: BACKOFF_FACTOR,
-                                interval_randomness: INTERVAL_RANDOMNESS, interval: INTERVAL, exceptions: RETRY_EXCEPTIONS
-        faraday.use Faraday::Adapter::NetHttp
-      end
+      Thread.current["#{THREAD_PREFIX}.connection"] = build_connection if creds_changed?
```

Comment:
Is this intentional: `chartmogul_ruby.connection.connection` ?
What about instead - when you change configuration, you `nil` this thread variable? Would it make simpler?
Ie. invalidate the cached connection when setting credentials.